mysql-test/suite/auth_sec/include/acl_tables_row_locking_test.inc

```diff
@@ -0,0 +1,721 @@
+--echo # We create new users so that we can populate rows in all acl tables.
+--echo # We avoid using 'root' user here, because that might affect execution
+--echo # of other tests. Also, we would need more than one users for below
+--echo # test cases.
+--echo #
+
+--let $SYSSE=`SELECT @@default_dd_system_storage_engine`
+
+CREATE TABLE t1 (f1 INT) ENGINE=InnoDB;
+DELIMITER //;
+CREATE PROCEDURE proc1() BEGIN SELECT * FROM t1; END //
+DELIMITER ;//
+--echo # User accounts used for testing.
+CREATE USER u1;
+GRANT SELECT ON mysql.* TO u1;
+GRANT SELECT ON performance_schema.* TO u1;
+GRANT LOCK TABLES  ON *.* TO u1;
+GRANT RELOAD ON *.* TO u1;
+
+CREATE USER writer1;
+GRANT SELECT, INSERT, UPDATE, DELETE ON mysql.* TO writer1;
+GRANT SELECT ON performance_schema.* TO writer1;
+GRANT CREATE, DROP ON *.* TO writer1;
+GRANT CREATE USER ON *.* TO writer1;
+
+CREATE USER writer2;
+GRANT SELECT, INSERT, UPDATE, DELETE ON mysql.* TO writer2;
+GRANT SELECT ON performance_schema.* TO writer2;
+GRANT CREATE, DROP ON *.* TO writer2;
+GRANT CREATE USER ON *.* TO writer2;
+GRANT SYSTEM_VARIABLES_ADMIN ON *.* TO writer2;
+GRANT REFERENCES ON mysql.user TO writer2;
+
+--echo # Populate all ACL tables with some rows for u2.
+--echo # This user is used for checking purpose.
+CREATE USER u2 PASSWORD HISTORY 3;
+GRANT SELECT ON test.* TO u2;
+GRANT SELECT ON test.t1 TO u2;
+GRANT SELECT (f1) ON test.t1 TO u2;
+GRANT EXECUTE ON PROCEDURE test.proc1 TO u2;
+GRANT PROXY ON writer1 TO u2;
+GRANT BACKUP_ADMIN ON *.* to u2;
+GRANT writer1 TO u2;
+SET DEFAULT ROLE writer1 TO u2;
+ALTER USER u2 IDENTIFIED BY 'newpass1';
+
+--echo # Verify that all ACL tables has some rows for u2.
+SELECT User FROM mysql.user WHERE User='u2';
+SELECT User FROM mysql.db WHERE User='u2';
+SELECT User FROM mysql.tables_priv WHERE User='u2';
+SELECT User FROM mysql.columns_priv WHERE User='u2';
+SELECT User FROM mysql.procs_priv WHERE User='u2';
+SELECT User FROM mysql.proxies_priv WHERE User='u2';
+SELECT User FROM mysql.global_grants WHERE User='u2';
+SELECT User FROM mysql.password_history WHERE User='u2';
+SELECT User FROM mysql.default_roles WHERE User='u2';
+SELECT To_user FROM mysql.role_edges WHERE To_user='u2';
+
+--echo # Backup user 'u2' and rename it as 'u2x'. This row
+--echo # is used to test DML on mysql.user.
+CREATE TABLE user_bk AS SELECT * FROM mysql.user WHERE User='u2';
+UPDATE user_bk SET user='u2x' WHERE user LIKE 'u2';
+ALTER TABLE user_bk ENGINE=InnoDB;
+CREATE TABLE user_copy AS SELECT * FROM mysql.user LIMIT 0;
+ALTER TABLE user_copy ENGINE=InnoDB;
+
+if ($SYSSE == "InnoDB")
+{
+let query_row_locks= SELECT CONCAT(OBJECT_SCHEMA,'.',OBJECT_NAME)
+  tab,INDEX_NAME,LOCK_MODE, 'Acquired row lock on u2/u2x' AS Has_lock
+  FROM performance_schema.data_locks
+  WHERE LOCK_TYPE='RECORD' AND
+  (LOCATE('u2',LOCK_DATA)>0 OR LOCATE('u2x',LOCK_DATA)>0);
+}
+if ($SYSSE == "RocksDB")
+{
+let query_row_locks= SELECT `KEY`, MODE FROM INFORMATION_SCHEMA.ROCKSDB_LOCKS
```

Comment:
looks like output 'KEY' may cause testcase flicky due to index id maybe different.

 SELECT `KEY`, MODE FROM INFORMATION_SCHEMA.ROCKSDB_LOCKS
 WHERE COLUMN_FAMILY_ID=0;
 KEY	MODE
-000001040000000000000001	X
+0000010b0000000000000001	X

Reply:
the first 4 bytes in these key is myrocks key index id --- different key has different index id to identity in myrocks.
such as for "000001040000000000000001", its key index id(in myrocks) is 0x00000104.  When a new table is created, all key will be assigned a  index id value and whose value is mono-increased.

For this case, we can't always assume that when the user_copy table is created at exactly order.  sometime, other testcass may executed before this testcase and will take these low key index id value.

maybe mark first 4 bytes?